From e8b298dab40738598a2b984e6b8689d5a45add0f Mon Sep 17 00:00:00 2001
From: Ikey Doherty <ikey.doherty@gmail.com>
Date: Sat, 7 Jun 2014 18:47:25 +0100
Subject: [PATCH 7/7] build: Ensure packages wont depend on themselves

In some instances a package may end up depending on itself due to
differences in lib64 and lib resolving of dependencies. This commit
ensures lib and lib64 are both checked if required and ensures any
part of the package cannot depend on itself (causing circular dependencies)
---
 pisi/operations/build.py | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pisi/operations/build.py b/pisi/operations/build.py
index 7dd54a2..0a54da5 100644
--- a/pisi/operations/build.py
+++ b/pisi/operations/build.py
@@ -942,10 +942,12 @@ class Builder:
 
                     # Ensure its not our own!
                     tpath = util.join_path(installdir, dep)
-                    tpath2 = tpath.replace("/lib/", "/lib64/")
+                    if "/lib64/" in tpath:
+                        tpath2 = tpath.replace("/lib64/", "/lib/")
+                    else:
+                        tpath2 = tpath.replace("/lib/", "/lib64/")
                     if os.path.exists(tpath) or os.path.exists(tpath2):
                         continue
-
                     if "lib64" in dep:
                         dep2 = dep.replace("/lib64/", "/lib/")
                     else:
-- 
1.9.1

